homeassistant/components/todoist/calendar.py

```diff
@@ -258,15 +259,15 @@ def handle_new_task(call: ServiceCall) -> None:
     )
 
 
-def _parse_due_date(data: dict, gmt_string) -> datetime | None:
-    """Parse the due date dict into a datetime object."""
-    # Add time information to date only strings.
-    if len(data["date"]) == 10:
-        return datetime.fromisoformat(data["date"]).replace(tzinfo=dt.UTC)
```

Comment:
This was the main issue, it ignored the user's timezone and parsed the due date as UTC with time information.